### DISCUSSION TITLE
Another SMTP failed to connect error

### DISCUSSION THREAD

**Discussion by MrBlackRDS** [PHPMailer/PHPMailer#2486] [Q&A]
I am receiving this error when adding my smtp information-

```
SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting
Connection: opening to smtp.office365.com:587, timeout=30, options=array()
Connection failed. Error #2: stream_socket_client(): Unable to connect to smtp.office365.com:587 (Network is unreachable) [/home/****/client.****.co/application/vendor/phpmailer/phpmailer/src/SMTP.php line 387]
SMTP ERROR: Failed to connect to server: Network is unreachable (101)
SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting
```
I reached out to Microsoft, to which they said they cannot provide support for 3rd party apps. They did mention trying this - 

```php
Get the credential
$credential = Get-Credential
 
Define the Send-MailMessage parameters
$mailParams = @{
    SmtpServer                 = 'smtp.office365.com'
    Port                       = '587' # or '25' if not using TLS
    UseSSL                     = $true ## or not if using non-TLS
    Credential                 = $credential
    From                       = 'sender@yourdomain.com'
    To                         = 'recipient@yourdomain.com', 'recipient@NotYourDomain.com'
    Subject                    = "SMTP Client Submission - $(Get-Date -Format g)"
    Body                       = 'This is a test email using SMTP Client Submission'
    DeliveryNotificationOption = 'OnFailure', 'OnSuccess'
}
```
I use Office 365 Business. Not sure if that helps figure out whats going on. Can anyone help me out?

**Comment by Synchro** [comment]
The language you seem to be writing code in is not PHP!

Beyond that, the problem here is clearly stated in the error message:

> Network is unreachable

This probably means that your hosting provider is blocking outbound SMTP. Diagnosing this is covered in great detail in the troubleshooting guide that was also linked from the error message.

First rule of debugging: *read the error message*.

**Comment by decomplexity** [comment]
Microsoft are suggesting use of Powershell commands to send email, but note that the Send-MailMessage cmdlet is deprecated and obsolete anyway. 
To Synchro's point, it is worth checking that your web hosting provider is not _GoDaddy_ or one of the many in _GoDaddy_'s stable.  Any web hosting provider using _GoDaddy_'s services will have outgoing email routed though a localhost server. If you are trying to use PHPMailer to send via a Microsoft 365 account MTA, it will be blocked. There is no way round this apart from switching to a VPS offering or switching provider.
